src/App.tsx

```diff
@@ -1,55 +1,97 @@
+      .includes(normalizeQuery(query)));
+
+  return filteredPeople;
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [apliedQuery, setApliedQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const refOfPost = useRef<HTMLInputElement | null>(null);
+  const filteredPeople = getFilteredPeople(apliedQuery);
+  const applyQuery = useCallback(debounce(setApliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleInputOnBlur = () => {
+    setTimeout(() => {
+      setIsFocused(false);
+    }, 100);
+  };
+
+  const handleButtonOnClick = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setIsFocused(true);
+  };
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? (`${selectedPerson.name} (${selectedPerson.born} = ${selectedPerson.died})`)
+          : ('No selected person')}
       </h1>
 
       <div className="dropdown is-active">
         <div className="dropdown-trigger">
           <input
             type="text"
+            value={query}
+            onChange={handleQueryChange}
+            onFocus={() => setIsFocused(true)}
+            onBlur={handleInputOnBlur}
+            ref={refOfPost}
             placeholder="Enter a part of the name"
             className="input"
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
+          <label className="dropdown-content">
+            {(query === apliedQuery)
+            && isFocused
+            && filteredPeople.length > 0
+            && (filteredPeople.map(person => (
```

Comment:
It looks too complicated. Create a variable for this condition or try to simplify it